### ISSUE TITLE
Get the top line and the bottom line on resizing

### ISSUE THREAD

**Issue by BDisp** [migueldeicaza/gui.cs#335]
I would like to launch a challenge to help me how to get the top line and the bottom line when the console size is smaller than the buffer size, that is, when the vertical scroll bar is visible. My intention is to make possible that both the menu and the status bar are always visible, regardless of the size of the console window. I've tried with the GetConsoleScreenBufferInfo function but it always returns the initial size of the console, regardless of the size change. Thanks.

**Comment by tig** [comment]
Another vote for this here: https://github.com/migueldeicaza/gui.cs/issues/355

The specific suggestion being: 

Add an event to `View` (e.g. `OnResize`) that gets called when the `View` is resized. All sorts of goodness could be implemented here, including getting the top & bottom line.

**Comment by BDisp** [comment]
@tig Believe me. This does not solve it. Not even the kernel libraries help in this case. The only one that can help in any way, obtains the size in pixels and it is not easy to obtain metrics to convert them to columns and rows, as well as based on the font size. Although it may be possible, it is not very easy to manage with all these constraints.

**Comment by tig** [comment]
Maybe we're not thinking the same way.

If I create a full size window, like this:

```csharp
            var win = new Window(_applicationData.Title)
            {
                X = 0,
                Y = 0, 
                // By using Dim.Fill(), it will automatically resize without manual intervention
                Width = Dim.Fill(),
                Height = Dim.Fill()
            };

            Application.Top.Add(win);
```

It shows a border all the way around it, and the bottom gets redrawn correctly when the window is resized.

Thus somewhere in `gui.cs` there is code that determines where the bottom of the console is on a resize.

**Comment by BDisp** [comment]
@tig Of course there is and gui.cs takes care of that, but it only works well if the size is being increased and not when it is shrinking. In cmd.exe properties you can see the actual size of the shrunk window but with the kernel library it only returns the size of the buffer and not just the actual size of the window. In addition, if the window is shrunk and the scroll bar is present in the middle of the window, there is no way to obtain the position of the top and bottom of the window. In Windows Forms this is already more feasible but not in consoles.

**Comment by tig** [comment]
Since I'm new to this, I'm not sure what you mean by the "kernel library".

**Comment by BDisp** [comment]
Like this one:
https://github.com/migueldeicaza/gui.cs/blob/72403fbd21731831bd672c8508d7a231561da3a7/Terminal.Gui/Drivers/WindowsDriver.cs#L358

**Comment by tig** [comment]
@BDisp where are you on this concept?

**Comment by BDisp** [comment]
If you question about my experience, it's not much. I have already thought of using other kernel functions to capture the entire resize of the window but only deal with pixels and converting them to cols and rows based on the font size is not always concise and proportional. I've worked on it a long time before, but not today. But it would be great to have the same behavior that I got for Linux, so with the OS that I have less knowledge :-)
